config/initializers/devise.rb

```diff
@@ -73,8 +73,11 @@ def valid?
   config.warden do |manager|
     manager.default_strategies(scope: :user).unshift :ldap_authenticatable if Devise.ldap_authentication
     manager.default_strategies(scope: :user).unshift :pam_authenticatable  if Devise.pam_authentication
-    manager.default_strategies(scope: :user).unshift :two_factor_authenticatable
-    manager.default_strategies(scope: :user).unshift :two_factor_backupable
+    manager.default_strategies(scope: :user).unshift :database_authenticatable
```

Comment:
Ah, if I `unshift` it *after* the other stuff, it gets the first position, which is wrong. I see.